### PR TITLE
Fix LGTM warnings/errors

### PR DESCRIPTION
**Pull request by jelly**
This PR fixes a bunch of warnings generated by LGTM and accumulated over the years https://lgtm.com/projects/g/cockpit-project/cockpit/alerts/?mode=list&severity=warning&lang=javascript

Ideally LGTM would complain instead of turn green in PR's.

### CODE REVIEW

File: pkg/lib/cockpit.js [hunk 1]
```diff
@@ -2524,7 +2524,7 @@ function factory() {
                 href = href.substr(1);
 
             const pos = href.indexOf('?');
-            let first = href;
+            let first;
             if (pos === -1)
                 first = href;
```
Comment (martinpitt) [inline pkg/lib/cockpit.js:2529]: This is better written as

    const first = (pos === -1) ? href : href.substr(0, pos);

**Review by martinpitt — APPROVED** [review]
Thanks! Still some test failures, not sure if they are related or just need a retry. Small issue in case you need to re-push.

**Comment by jelly** [comment]
> Thanks! Still some test failures, not sure if they are related or just need a retry. Small issue in case you need to re-push.

Fixed, re-pushed. Only packit rawhide fails due to python 3.11 which is expected.

**Review by martinpitt — APPROVED** [review]
LGTM! :wink:

**Pull request merged**